app/helpers/course/assessment/answer/programming_helper.rb

```diff
@@ -123,4 +155,45 @@ def build_annotation_discussion(line_discussion_cell, discussion_topic)
 
     line_discussion_cell.inner_html = html
   end
+
+  # Return a hash of the first failing test case and its test result
+  #
+  # @param [Hash] test_cases_and_results_of_type A hash of test cases and results keyed by type
+  # @return [Hash] the failed test case and result, nil if all tests passed
+  def get_first_failed_test(test_cases_and_results_of_type)
+    test_cases_and_results_of_type.each do |test_case, test_result|
```

Comment:
hound was correct, code has been changed so the hash key is really the test case and not the symbol `:test_case`.